src/module.c

```diff
@@ -1022,6 +1162,15 @@ int TSDB_mget(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
             RedisModule_ReplyWithArray(ctx, 0);
         }
         ReplyWithSeriesLastDatapoint(ctx, series);
+
+        // undo latest sample on LATEST param
+        if (should_finalize_last_bucket) {
+            if (srcSeries->totalSamples > 0) {
+                CompactionRule *rule = find_rule(srcSeries->rules, series->keyName);
+                SeriesDelRange(series, rule->startCurrentTimeBucket, rule->startCurrentTimeBucket);
```

Comment:
@OfirMos why do we need to delete when LATEST is called?

Reply:
@danni-m Yes, cause we closed the bucket while there might still arrive more samples that are belong to this bucket,
And as part of the latest implementation we wrote the last bucket to the downsampled series.